app/main.py

```diff
@@ -1,25 +1,86 @@
 from abc import ABC
+from typing import Any, Type
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __get__(self, instance: Any, owner: Type) -> int:
+        return getattr(instance, self.name)
+
+    def __set__(self, instance: Any, value: int) -> None:
+        if not self.min_amount <= value <= self.max_amount:
+            raise ValueError(f"{self.name} must be between \
+                {self.min_amount} and {self.max_amount}")
+        setattr(instance, self.name, value)
+
+    def __set_name__(self, owner: Type, name: str) -> None:
+        self.name = name
 
 
 class Visitor:
-   pass
+    def __init__(
+        self,
+        name: str,
+        age: int,
+        weight: int,
+        height: int
+    ) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    def __init__(
+        self,
+        age: int,
+        height: int,
+        weight: int
+    ) -> None:
+        self.age = age
+        self.height = height
+        self.weight = weight
+
+    def validate(self, visitor: Visitor) -> bool:
+        age = visitor.age
+        height = visitor.height
+        weight = visitor.weight
+        return (
+            self.age.min_amount <= age <= self.age.max_amount
+            and self.height.min_amount <= height <= self.height.max_amount
+            and self.weight.min_amount <= weight <= self.weight.max_amount
+        )
 
 
 class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    def __init__(self) -> None:
+        super().__init__(
+            age=IntegerRange(4, 14),
+            height=IntegerRange(80, 120),
+            weight=IntegerRange(20, 50))
 
 
 class AdultSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    def __init__(self) -> None:
+        super().__init__(
+            age=IntegerRange(14, 60),
+            height=IntegerRange(120, 220),
+            weight=IntegerRange(50, 120))
 
 
 class Slide:
-    pass
+    def __init__(
+        self,
+        name: str,
+        limitation_class: ChildrenSlideLimitationValidator
+            | AdultSlideLimitationValidator
```

Comment:
Change your type annotation to indicate that you expect to obtain a subclass of `SlideLimitationValidator` class:
```suggestion
        limitation_class: type[SlideLimitationValidator]
```